Mathlib/Data/Matrix/Rank.lean

```diff
@@ -114,15 +207,47 @@ theorem rank_submatrix_le [StrongRankCondition R] [Fintype m] (f : n → m) (e :
     LinearEquiv.range, Submodule.map_top]
   exact Submodule.finrank_map_le _ _
 
-theorem rank_reindex (e₁ : l ≃ m) (e₂ : n ≃ o) (A : Matrix l n R) :
-    rank (reindex e₁ e₂ A) = rank A := by
+@[simp]
+theorem rank_reindex [Fintype n₀] (em : m ≃ m₀) (en : n ≃ n₀) (A : Matrix m n R) :
+    rank (A.reindex em en) = rank A := by
   rw [rank, rank, mulVecLin_reindex, LinearMap.range_comp, LinearMap.range_comp,
     LinearEquiv.range, Submodule.map_top, LinearEquiv.finrank_map_eq]
 
 @[simp]
-theorem rank_submatrix (A : Matrix l n R) (e₁ : m ≃ l) (e₂ : o ≃ n) :
-    rank (A.submatrix e₁ e₂) = rank A :=
-  rank_reindex e₁.symm e₂.symm A
+theorem rank_submatrix [Fintype n₀] (A : Matrix m n R) (em : m₀ ≃ m) (en : n₀ ≃ n) :
+    rank (A.submatrix em en) = rank A := by
+  simpa only [reindex_apply] using rank_reindex em.symm en.symm A
+
+theorem lift_cRank_reindex {n : Type un} (A : Matrix m n R) (em : m ≃ m₀) (en : n ≃ n₀) :
```

Comment:
```suggestion
theorem lift_cRank_reindex (A : Matrix m n R) (em : m ≃ m₀) (en : n ≃ n₀) :
```
since this variable already exists

Reply:
It was the shortest way I could see to get rid of the `Fintype n` floating around.

Reply:
Once `rank` is redefined, the global `Fintype` can probably just disappear, since most lemma statements won't need it.